02.async-await/AsyncAwait.Task2.CodeReviewChallenge/Controllers/HomeController.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using System.Diagnostics;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using AsyncAwait.Task2.CodeReviewChallenge.Models;
+using AsyncAwait.Task2.CodeReviewChallenge.Models.Support;
+using AsyncAwait.Task2.CodeReviewChallenge.Services;
+
+namespace AsyncAwait.Task2.CodeReviewChallenge.Controllers
+{
+    public class HomeController : Controller
+    {
+        private readonly IAssistant _assistant;
+
+        private readonly IPrivacyDataService _privacyDataService;
+
+        public HomeController(IAssistant assistant, IPrivacyDataService privacyDataService)
+        {
+            _assistant = assistant ?? throw new ArgumentNullException(nameof(assistant));
+            _privacyDataService = privacyDataService ?? throw new ArgumentNullException(nameof(privacyDataService));
+        }
+
+        public ActionResult Index()
+        {
+            return View();
+        }
+
+        public ActionResult Privacy()
+        {
+            ViewBag.Message = _privacyDataService.GetPrivacyDataAsync().Result;
+            return View();
+        }
+
+        public async Task<IActionResult> Help()
+        {
+            ViewBag.RequestInfo = await _assistant.RequestAssistanceAsync("guest").ConfigureAwait(false);
```

Comment:
No need to use `.ConfigureAwait(false)` here. You should use `ConfigureAwait` in library lvl to prevent deadlock on synchronous task execution